### PR TITLE
Add ResNet, AlexNet, and VGG model definitions and model zoo

### PR DESCRIPTION
**Pull request by colesbury**
You can create a ResNet-18 model pre-trained on ImageNet with:

```python
import torchvision.models
resnet = torchvision.models.resnet18(pretrained=True)
```

### CODE REVIEW

**Comment by colesbury** [comment]
The ResNet models were trained on ILSVRC2012. They were trained with the default options using https://github.com/pytorch/examples/tree/master/imagenet. (batch size 256, 90 epochs, lr=0.1 decay by 10 every 30 epochs). I think I used 4 GPUs, but it might have been 8.

Accuracy on the validation set is in the table here:
https://pytorch.org/docs/stable/torchvision/models.html

You can verify the accuracy on the validation set by running the example imagenet script with:
```
python main.py -a resnet18 --evaluate --pretrained [path to imagenet]
```

You can compute the accuracy on the training set by swapping your train and val directories and running the above command.
